core/shared/src/main/scala/zio/FiberRefs.scala

```diff
@@ -234,9 +289,13 @@ object FiberRefs {
      */
     def diff(oldValue: FiberRefs, newValue: FiberRefs): Patch = {
       val (removed, patch) = newValue.fiberRefLocals.foldLeft[(FiberRefs, Patch)](oldValue -> empty) {
-        case ((fiberRefs, patch), (fiberRef, ((_, newValue, _) :: _, _))) =>
-          fiberRefs.get(fiberRef) match {
-            case Some(oldValue) =>
+        case ((fiberRefs, patch), (fiberRef, Value(StackEntry(_, newValue, _) :: _, _))) =>
+          fiberRefs.getOrNull(fiberRef) match {
+            case null =>
+              fiberRefs.delete(fiberRef) -> patch.combine(
```

Comment:
Why do we need to `fiberRefs.delete` if `fiberRefs.getOrNull(fiberRef)` returns `null`? 🤔